index.js

```diff
@@ -1,10 +1,56 @@
+var HAS_PROCESS_ENV_NODE_ENV = typeof process !== 'undefined' && Object.prototype.hasOwnProperty(process, 'env') && Object.prototype.hasOwnProperty(process.env, 'NODE_ENV');
+var IS_PRODUCTION = HAS_PROCESS_ENV_NODE_ENV && process.env.NODE_ENV === 'production';
+var Immutable = require('immutable');
+
+function isImmutable(value) {
+    // immutable v4 added `isImmutable`
+    if (Immutable.hasOwnProperty(isImmutable)) {
+        return Immutable.isImmutable(value);
+    } else {
+        return Immutable.Map.isMap(value) ||
```

Comment:
these are the only supported immutable types?

Reply:
The others don't have `isX` methods